### PR TITLE
Sheet gene

### PR DESCRIPTION
**Pull request by michaone007**

### CODE REVIEW

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 1]
```diff
@@ -65,6 +69,68 @@ public static TriviaQuestion generateQuestionWithData(ArrayList<HashMap<String,
...
+        HashMap<String, Object> questionHashFromData = GeneratorHelper.maxBuildQuestion(data, questionKey, answerKey, num);
+        String right = (String) questionHashFromData.get("rightAnswer");
+        ArrayList<String> answers = (ArrayList<String>) questionHashFromData.get("answers");
+
+        TriviaQuestion question = new TriviaQuestion("someKey", q, right, answers.get(0), answers.get(1), answers.get(2), answers.get(3));
+        return question;
+    }
+
+    public static HashMap<String, Object> maxBuildQuestion(ArrayList<HashMap<String, Object>> data,  String questionKey,  String answerKey, int num) {
+        ArrayList<HashMap<String, String>> d = new ArrayList<>();
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: change var name

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 2]
```diff
@@ -65,6 +69,68 @@ public static TriviaQuestion generateQuestionWithData(ArrayList<HashMap<String,
...
+        Collections.shuffle(data);
+        int i = 0,k=1;
+        while (set.size() < 4 ){
+            HashMap<String,Object> answerHash = data.get(i);
+            String answer = (String)answerHash.get(questionKey);
+            String answer2 = (String)answerHash.get(answerKey);
+                set.add(Integer.valueOf(answer2));
+                if(set.size()==k) {
+                    d.add(new HashMap<String, String>());
+                    d.get(d.size() - 1).put(answer,answer2);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: first build the hash and then add it to 'd'. 
hash map should be like that: 
{"name":"Barcelona", "number":"33"}

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 3]
```diff
@@ -78,6 +82,69 @@ public static TriviaQuestion generateQuestionWithData(ArrayList<HashMap<String,
...
+          if (set.size()==k) {
+                HashMap helper = new HashMap();
+                helper.put("name", name);
+                helper.put("number", number);
+                fourAnswers.add(helper);
+              k++;
+            }
+            i++;
+            if (i == data.size()) {
+                i = 1;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java:115]: if (i == data.size()) setting i = 1 would not help us as we will still enter an infinite loop.. in this case we should push dummy answers ("XXX") with negative values for max mode and high positive value for min mode. 
Nevertheless, we should not reach this case anyway.

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java [hunk 4]
```diff
@@ -78,6 +82,69 @@ public static TriviaQuestion generateQuestionWithData(ArrayList<HashMap<String,
...
+    public static TriviaQuestion maxGenerateQuestionWithData(ArrayList<HashMap<String, Object>> data, String q, String questionKey, String answerKey, int num) {
+        Log.d(MainActivity.TAG, "data = " + data);
+        HashMap<String, Object> questionHashFromData = GeneratorHelper.maxBuildQuestion(data, questionKey, answerKey, num);
+        String right = (String) questionHashFromData.get("rightAnswer");
+        ArrayList<String> answers = (ArrayList<String>) questionHashFromData.get("answers");
+
+        return new TriviaQuestion("someKey", q, right, answers.get(0), answers.get(1), answers.get(2), answers.get(3), false);
+    }
+
+    public static HashMap<String, Object> maxBuildQuestion(ArrayList<HashMap<String, Object>> data, String questionKey, String answerKey, int num) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: what is `num`? is it the flag for max \ min? if so, please change its name and make it boolean.

Comment (michaone007) [reply app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java]: yes ill change..

File: app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java (hunk 3 above)
Comment (michaone007) [reply app/src/main/java/com/example/zakiva/santa/Helpers/GeneratorHelper.java:115]: like you did in your function. no problem.

**Pull request merged**